File: frontend/src/modules/marketplace/types/marketplace.ts
export interface CommercialAssetItem {
  id: string;
  name: string;
  type: string;
  priceUsd: number;
  publisherId: string;
  publisherName: string;
  rating: number;
  downloadsCount: number;
  status: string;
  description: string;
  createdAt: string;
}

export interface CommercialSubscriptionItem {
  id: string;
  workspaceId: string;
  tier: string;
  seats: number;
  status: string;
  createdAt: string;
}

export interface LicenseItem {
  id: string;
  workspaceId: string;
  assetId: string;
  licenseKey: string;
  seatsAllocated: number;
  status: string;
}

export interface ReviewItem {
  id: string;
  assetId: string;
  userName: string;
  rating: number;
  comment: string;
  isVerifiedPurchase: boolean;
  createdAt: string;
}

File: backend/app/schemas/marketplace.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class MarketplaceAssetResponse(BaseModel):
    id: str
    name: str
    type: str
    price_usd: float
    publisher_id: str
    publisher_name: Optional[str] = "Enterprise AI Labs"
    rating: float
    downloads_count: int
    status: str
    description: Optional[str]
    created_at: datetime

    class Config:
        from_attributes = True

class SubscriptionResponse(BaseModel):
    id: str
    workspace_id: str
    tier: str
    seats: int
    status: str
    created_at: datetime

    class Config:
        from_attributes = True

class InvoiceResponse(BaseModel):
    id: str
    workspace_id: str
    amount_usd: float
    status: str
    paid_at: datetime

    class Config:
        from_attributes = True

class LicenseResponse(BaseModel):
    id: str
    workspace_id: str
    asset_id: str
    license_key: str
    seats_allocated: int
    status: str

    class Config:
        from_attributes = True

class ReviewResponse(BaseModel):
    id: str
    asset_id: str
    user_name: str
    rating: int
    comment: str
    is_verified_purchase: bool
    created_at: datetime

    class Config:
        from_attributes = True

class CheckoutSessionRequest(BaseModel):
    asset_id: Optional[str] = None
    tier: Optional[str] = "Pro"
    seats: int = 10

class CheckoutSessionResponse(BaseModel):
    checkout_url: str
    session_id: str
